utils_cv/detection/dataset.py

```diff
@@ -0,0 +1,220 @@
+            image_folder: the name of the image folder
+            annotation_folder: the name of the annotation folder
+        """
+
+        self.root = Path(root)
+        self.transforms = transforms
+        self.image_folder = image_folder
+        self.annotation_folder = annotation_folder
+
+        self.ims = list(sorted(os.listdir(self.root / self.image_folder)))
+        self.annotations = list(
+            sorted(os.listdir(self.root / self.annotation_folder))
+        )
+        self.categories = self._get_categories()
+
+    def _get_categories(self) -> List[str]:
+        """ Parses all Pascal VOC formatted annoatation files to extract all
+        possible categories. """
+        categories = ["__background__"]
+        for annotation_path in self.annotations:
+            annotation_path = self.root / "annotations" / str(annotation_path)
+            tree = ET.parse(annotation_path)
+            root = tree.getroot()
+            objs = root.findall("object")
+            for obj in objs:
+                category = obj[0]
+                assert category.tag == "name"
+                categories.append(category.text)
+        return list(set(categories))
+
+    def get_image_features(
+        self, idx: int = None, rand: bool = False
+    ) -> Tuple[List[List[int]], List[str], str]:
+        """ Choose and get image from dataset.
+
+        This function returns all the data that is needed to effectively
+        visualize an image from the dataset.
+
+        Args:
+            idx: The image index to get the features of
+            rand: randomly select image
+
+        Raises:
+            Exception is idx is not None and rand is set to True
+
+        Returns:
+            A tuple of boxes, categories, image path
+        """
+        if (idx is not None) and (rand is True):
+            raise Exception("idx cannot be set if rand is set to True.")
+        if idx is None and rand is False:
+            rand = True
+        if rand:
+            idx = randrange(len(self.ims))
+        boxes, labels, im_path = self._get_im_data(idx)
+        return (boxes, [self.categories[label] for label in labels], im_path)
+
+    def split_train_test(
+        self, train_ratio: float = 0.8
+    ) -> Tuple[Dataset, Dataset]:
+        """ Split theis dataset into a training and testing set
+
+        Args:
+            train_ratio: the amount of images to use for training (the rest
+            will be used for testing.
+
+        Return
+            A training and testing dataset in that order
+        """
+        test_num = math.floor(len(self) * (1 - train_ratio))
+        indices = torch.randperm(len(self)).tolist()
+        self.set_transform(get_transform(train=True))
+        train = Subset(self, indices[:-test_num])
+        self.set_transform(get_transform(train=False))
+        test = Subset(self, indices[-test_num:])
+        return train, test
+
+    def set_transform(self, transforms: List[object]) -> None:
```

Comment:
Do we need this function? If it is such that the naming is similar to fastai then I understand. It's just that transforms is public and could be set directly, this function isn't doing anything on top.